index.js

```diff
@@ -335,53 +294,24 @@ class TuyaDevice extends EventEmitter {
 
         // Parse response data
         this.client.on('data', data => {
-          debug('Received response');
-          debug(data.toString('hex'));
+          debug(`Received data: ${data.toString('hex')}`);
 
-          // Response was received, so stop waiting
-          clearTimeout(this._sendTimeout);
+          let packets;
 
-          let dataRes;
           try {
-            dataRes = Parser.parse(data);
+            packets = this.device.parser.parse(data);
           } catch (error) {
             debug(error);
             this.emit('error', error);
             return;
           }
 
-          data = dataRes.data;
-
-          if (typeof data === 'object') {
-            debug('Parsed response data:');
-            debug(data);
-          } else if (typeof data === 'undefined') {
-            if (dataRes.commandByte === 0x09) { // PONG received
-              debug('Pong', this.device.ip);
-              return;
-            }
-
-            if (dataRes.commandByte === 0x07) { // Set succeeded
-              debug('Set succeeded.');
-              return;
-            }
-
-            debug(`Undefined data with command byte ${dataRes.commandByte}`);
-          } else { // Message is encrypted
-            data = this.device.cipher.decrypt(data);
```

Comment:
Really happy with this direction, glad you could find a better home for crypto functions